docs/docs/developers/tutorials/crowdfunding/donations.md

```diff
@@ -0,0 +1,202 @@
+
+The `ValueNote` type is in the top-level of the Aztec.nr framework, namely [noir-projects/aztec-nr](https://github.com/AztecProtocol/aztec-packages/blob/#include_aztec_version/noir-projects/aztec-nr/value-note/src/value_note.nr). Like before, you'll need to add the crate to Nargo.toml
+
+(See [here](https://docs.aztec.network/developers/contracts/resources/dependencies) for common dependencies).
+
+---
+
+Back in main.nr, reference `use` of the type
+
+```rust
+use dep::value_note::value_note::ValueNote;
+```
+
+Now complete the initializer by setting the storage variables with the parameters:
+
+#include_code init /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr rust
+
+You can compile the code so far with `aztec-nargo compile`.
+
+### 2. Taking private donations
+
+#### Checking campaign duration against the timestamp
+
+To check that the donation occurs before the campaign deadline, we must access the public `timestamp`. It is one of several [Public Global Variables](https://docs.aztec.network/developers/contracts/references/globals#public-global-variables).
+
+Declare an Aztec function that is public and internal
+
+```rust
+#include_code deadline-header /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr raw
+    //...
+}
+```
+
+Read the deadline from storage and assert that the `timestamp` from this context is before the deadline
+
+#include_code deadline /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr rust
+
+---
+
+Since donations are to be private, the donate function will have the user's private context which has these [Private Global Variables](https://docs.aztec.network/developers/contracts/references/globals#private-global-variables). So from the private context there is a little extra to call the (public internal) `_check_deadline` function.
+
+```rust
+#include_code call-check-deadline /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr raw
+  //...
+}
+```
+
+Namely calling `enqueue` and passing the (mutable) context.
+
+Now conclude adding all dependencies to the `Crowdfunding` contract:
+
+#include_code all-deps /noir-projects/noir-contracts/contracts/crowdfunding_contract/src/main.nr rust
+
+Like before, you can find these and other `aztec::protocol_types` [here](https://github.com/AztecProtocol/aztec-packages/blob/#include_aztec_version/noir-projects/noir-protocol-circuits/crates/types/src).
+
+
+#### Interfacing with another contract
+
+The token being used for donations is stored simply as an `AztecAddress` (named `donation_token`). so to easily use it as a token, we let the compiler know that we want the address to have a Token interface. Here we will use a maintained example Token contract.
+
+Add this `Token` contract to Nargo.toml: `aztec = { git="https://github.com/AztecProtocol/aztec-packages/", tag="#include_aztec_version", directory="noir-projects/noir-contracts/contracts/token_contract" }`
```

Comment:
Should be (can't suggest because the three ``` mess up formatting of the suggestion):


Add this `Token` contract to Nargo.toml: 
```
token = { git="https://github.com/AztecProtocol/aztec-packages/", tag="#include_aztec_version", directory="noir-projects/noir-contracts/contracts/token_contract" }
```

Reply:
it looks like you didn't update `aztec` to `token`